pkg/readiness/ready_tracker.go

```diff
@@ -46,7 +46,7 @@ import (
 
 var log = logf.Log.WithName("readiness-tracker")
 
-var crashOnFailureFetchingExpectations = flag.Bool("crash-on-failure-fetching-expectations", false, "Unless set (defaults to false), gatekeeper will ignore errors that occur when gathering expectations. This prevents bootstrapping errors from crashing Gatekeeper at the cost of increasing the risk Gatekeeper will under-enforce policy by serving before it has loaded in all policies. Enabling this will help prevent under-enforcement at the risk of crashing during startup for issues like network errors. Note that enabling this flag currently does not achieve the aforementioned effect since fetching expectations are set to retry until success so failures during fetching expectations currently do not occur.")
+var crashOnFailureFetchingExpectations = flag.Bool("crash-on-failure-fetching-expectations", false, "Unless set (defaults to false), gatekeeper will ignore errors when gathering expectations. This prevents bootstrapping errors from crashing Gatekeeper at the cost of increasing the risk Gatekeeper will under-enforce policy. Enabling this will help prevent under-enforcement at the risk of crashing during startup. Note that enabling this flag currently does not achieve the aforementioned effect since fetching expectations will retry until success.")
```

Comment:
> Note that enabling this flag currently does not achieve the aforementioned effect since fetching expectations will retry until success.

This says instead of crashing, fetching expectations will continue to retry until success. If so, then this flag is misleading as it starts with "crash". Is there plan to implement crash immediately on failure? If not, then should this be renamed to "retryOnFailureFetchingExpectations`? and when set to false, it will ignore on failure.

Reply:
Right now the flag description/name is accurate but moot.

The wording is appropriate since that is how the code works -- this flag basically controls whether errors fetching expectations are bubbled up to controller-manager (which would cause the program to crash). Currently errors just cause an infinite retry loop, meaning there are never any errors to bubble up.

If we ever want to add a maximum amount of retries for fetching expectations (could be useful both for users who want to fail open and for users who want to crash-on-failure to raise visibility of issues), then this flag would immediately be useful.

IMO if we don't want the flag exposed to users while its value is moot, we should leave the flag variable as a global "constant", set to `&false` and comment out the `flag.Bool()` bit that exposes the flag to users. This will retain the clarity we gained in how the code functions from the recent refactoring.

Reply:
Though, if we want to do that, we need to do that before the next release is cut to avoid breaking changes.

Reply:
> IMO if we don't want the flag exposed to users while its value is moot, we should leave the flag variable as a global "constant", set to &false and comment out the flag.Bool() bit that exposes the flag to users.

+1 on not exposing the flag until the value is not moot.

Reply:
@David-Jaeyoon-Lee can you please make the suggested changes soon? We plan to cut a release in a week. Thank you!

Reply:
I can open a PR to remove the flag for now if that's easier. Just let me know what you prefer. Thanks!

Reply:
This sounds like a good idea. I'll make the change by EOD today. Sorry for the wait. @ritazh